### PR TITLE
Add go module example and links

### PR DESCRIPTION
**Pull request by Joelkw**
Add another use case, which also demonstrates how to use non-capturing groups as well. 

## Test plan

Reviewed manually, docs change.

### CODE REVIEW

File: doc/code_insights/references/search_aggregations_use_cases.md [hunk 1]
```diff
@@ -55,6 +55,12 @@ See every email address hardcoded, by frequency
 /(\w+)\@sourcegraph\.com/
 ```
 
+### What are the most common go modules in use? 
+See the most common go modules in go.mod files
+```sgquery
+file:go.mod$ /(.+)\sv(?:.+)(?:\s\/\/\sindirect)?/
```
Comment (chwarwick) [inline doc/code_insights/references/search_aggregations_use_cases.md:61]: I believe the intent of the search was to exclude indirect additions to the go.mod, and this search would include them.

Without support for negative look aheads I think you would have to try to do a semver regex match and then check for an immediate \n to exclude the indirects.

This isn't the most friendly looking search but this excludes the indirects:
```
file:go.mod$ /(.+)\sv(?:(?:(?:[0-9]+)\.(?:[0-9]+)\.(?:[0-9]+)(?:-(?:[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)\n/
```

File: doc/code_insights/references/index.md [hunk 2]
```diff
@@ -3,7 +3,8 @@
 The following is a list of reference documents for [Code Insights](../index.md):
 
 - [Common use cases and recipes](common_use_cases.md)
+    - [Search results aggregations use cases and recipes](search_aggregations_use_cases.md)
```
Comment (chwarwick) [inline doc/code_insights/references/index.md:6]: Something off on the spacing here.  Is this supposed to be a sub item?

Comment (Joelkw) [reply doc/code_insights/references/index.md:6]: Yes, I'd like to to be a sub item of "recipes" pages and didn't want to alphabetize, but thanks for checking.

File: doc/code_insights/references/search_aggregations_use_cases.md (hunk 1 above)
Comment (Joelkw) [reply doc/code_insights/references/search_aggregations_use_cases.md:61]: Ah, nice catch!

Comment (Joelkw) [inline doc/code_insights/references/search_aggregations_use_cases.md:61]:
```suggestion
file:go.mod$ /(.+)\sv(?:.+)(?:\s\/\/\sindirect)?/
```

Or, to exclude indirect additions, use:
```sgquery
file:go.mod$ /(.+)\sv(?:(?:(?:[0-9]+)\.(?:[0-9]+)\.(?:[0-9]+)(?:-(?:[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)\n/
```

**Review by felixfbecker — APPROVED** [review]

**Comment by Joelkw** [comment]
not sure still relevant

**Pull request closed without merging**